b_cfn_sagemaker_endpoint/refresh/source/index.py

```diff
@@ -0,0 +1,68 @@
+import json
+import os
+import time
+from typing import Any, Dict
+import boto3
+
+
+class Counter:
+    def __init__(self, start: int = 0, step: int = 1):
+        self.value = start
+        self.__step = step
+
+    def next(self) -> int:
+        self.value += self.__step
+        return self.value
+
+
+invocation_counter = Counter()
+
+
+def handler(event: Dict[str, Any], context: Any) -> None:
+    invocation_number = invocation_counter.next()
+    print('Current invocation number:', invocation_number)
+    print(f'S3 event received: {json.dumps(event)}')
+
+    wait_time = float(os.environ['WAIT_TIME'])
+    endpoint_name = os.environ['SAGEMAKER_ENDPOINT_NAME']
+    endpoint_config_a_name = os.environ['SAGEMAKER_ENDPOINT_CONFIG_A_NAME']
+    endpoint_config_b_name = os.environ['SAGEMAKER_ENDPOINT_CONFIG_B_NAME']
+    print(
+        'Using the following environment variables: '
+        f'{wait_time=} '
+        f'{endpoint_name=} '
+        f'{endpoint_config_a_name=} '
+        f'{endpoint_config_b_name=} '
+    )
+
+    # Wait for any other bucket objects to be uploaded.
+    print('Waiting on standby...')
+    time.sleep(wait_time)
+
+    # Make sure that only the latest event handler invocation would be run.
+    latest_invocation_number = invocation_counter.value
```

Comment:
How it is possible to have such edge case that you are handling? I smell over-engineering :D